libraries/exoplayer/src/main/java/androidx/media3/exoplayer/offline/SegmentDownloader.java

```diff
@@ -107,7 +108,29 @@ public int compareTo(Segment other) {
    * @param executor An {@link Executor} used to make requests for the media being downloaded.
    *     Providing an {@link Executor} that uses multiple threads will speed up the download by
    *     allowing parts of it to be executed in parallel.
+   * @param maxMergedSegmentStartTimeDiffUs max merged size for each segment duration
    */
+  public SegmentDownloader(
+      MediaItem mediaItem,
+      Parser<M> manifestParser,
+      CacheDataSource.Factory cacheDataSourceFactory,
+      Executor executor,
+      long maxMergedSegmentStartTimeDiffUs
```

Comment:
nit: I think as the API value, millisecond is sufficient for this purpose. Can you please change the constructor parameter to `maxMergedSegmentStartTimeDiffMs`. Leave the final field in Us and then convert the param using `Util.msToUs(maxMergedSegmentStartTimeDiffMs)` when assigning to the final field.